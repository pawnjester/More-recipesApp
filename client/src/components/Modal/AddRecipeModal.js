import React, { Component } from 'react';
import { Button,Container, Modal, 
    ModalHeader, ModalBody, ModalFooter, Form, Label, Input, FormGroup, Col, FormText } from 'reactstrap';

class AddRecipeModal extends Component {
    constructor(props) {
        super(props);

        this.state = {
            name: '',

        }
        
        this.onNameChange = this.onNameChange.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
    }


    onNameChange(event) {
        this.setState({ [event.target.name]: event.target.value})
    }

    onSubmit() {
        alert(`Saving ${this.state.name}`)
    }
      render() {
        return (
            <Modal isOpen={this.props.isOpen} toggle={this.props.toggle}>
              <ModalHeader toggle={this.props.toggle}>Add A recipe</ModalHeader>
              <ModalBody>
                <Form>
                <FormGroup row>
                <Label for="exampleEmail" sm={4}>Name</Label>
                <Col sm={8}>
                  <Input 
                  type="text" 
                  name= "name" 
                  id="exampleEmail" 
                  value = {this.state.name}
                  onChange = {this.onNameChange}
                  placeholder="Enter the name" />
                </Col>
              </FormGroup>
              
              <FormGroup check row>
                <Col sm={{ size: 10, offset: 2 }}>
                  <Button  styles = "cursor:pointer" onClick= {this.onSubmit}>Add a recipe</Button>
                </Col>
              </FormGroup>
                </Form>
              </ModalBody>
            </Modal>
        )
      }
    }
    

    

export default AddRecipeModal;