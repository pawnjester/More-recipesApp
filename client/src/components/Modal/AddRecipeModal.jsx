import React, { Component } from 'react';
import toastr from 'toastr';
import {
  Button, Modal,
  ModalHeader, ModalBody, Form, Label, Input, FormGroup, Col, FormText
} from 'reactstrap';
import imageUpload from '../../helpers/imageUpload';
import validateInput from '../recipe/validations';

/* eslint-disable */
class AddRecipeModal extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: '',
      ingredients: '',
      method: '',
      imageUrl: '',
      cookingTime: 0,
      status: '',
      errors: {},
    };

    this.onNameChange = this.onNameChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.Upload = this.Upload.bind(this);
  }


  onNameChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  isValid() {
    const { errors, isValid } = validateInput(this.state);

    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }

  onSubmit(event) {
    event.preventDefault();
    // if (this.isValid()) {
    //   this.setState({ errors: {} });
    //   this.props.createRecipe(this.state).then((res) => {
    //     toastr.success('Recipe added');
    //     this.props.toggle();
    //   })

    // }
    if(this.state.errors !== {}) {
      this.setState({ errors: this.props.errors})
      console.log('~~~~~~~||||')
      this.props.toggle();
    } else if (this.isValid()){
      this.setState({ errors: {} });
      this.props.createRecipe(this.state).then((res) => {
        toastr.success('Recipe added');
        this.props.toggle();
      })
    }
  }

  Upload(images) {
    this.setState({ status: 'Uploading...' })
    imageUpload(images).then((response) => {
      const { body } = response
      const fileUrl = body.secure_url;

      if (fileUrl) {
        this.setState({
          imageUrl: fileUrl,
          status: 'Uploaded'
        })
      }
    })
  }

  render() {
    const { errors } = this.state;
    console.log('>>>>', errors);
    console.log('>>this.props', this.props.errors)
    return (
      <Modal isOpen={this.props.isOpen} toggle={this.props.toggle}>
        <ModalHeader toggle={this.props.toggle}>Add A recipe</ModalHeader>
        <ModalBody>
          <Form>
            <FormGroup row>
              <Label for="exampleEmail" sm={3}>Name</Label>
              <Col sm={9}>
                <Input
                  type="text"
                  name="name"
                  id="exampleEmail"
                  value={this.state.name}
                  onChange={this.onNameChange}
                  placeholder="Enter the name"
                />
                {errors.error || errors.name && <small style={{color: '#A43741' }}>{errors.name}</small>}
              </Col>
            </FormGroup>

            <FormGroup row>
              <Label for="exampleEmail" sm={3}>Ingredients</Label>
              <Col sm={9}>
                <Input
                  type="textarea"
                  name="ingredients"
                  id="exampleEmail"
                  value={this.state.ingredients}
                  onChange={this.onNameChange}
                  placeholder="Enter the Ingredients"
                  style={{ height: 150}}
                />
                {errors.error || errors.ingredients && <small style={{color: '#A43741' }}>{errors.ingredients}</small>}
              </Col>
            </FormGroup>

            <FormGroup row>
              <Label for="exampleEmail" sm={3}>method</Label>
              <Col sm={9}>
                <Input
                  type="textarea"
                  name="method"
                  id="exampleEmail"
                  value={this.state.method}
                  onChange={this.onNameChange}
                  placeholder="Enter the description"
                  style={{ height: 150}}
                />
                {errors.error || errors.method && <small style={{color: '#A43741' }}>{errors.method}</small>}
              </Col>
            </FormGroup>

            <FormGroup row>
              <Label for="exampleEmail" sm={3}>Preparation Time</Label>
              <Col sm={9}>
                <Input
                  type="text"
                  name="cookingTime"
                  id="exampleEmail"
                  value={this.state.cookingTime}
                  onChange={this.onNameChange}
                  placeholder="Enter the preparation time"
                />

                <Col sm={12}>
                  <Input
                  type="select"
                  name="select"
                  id="exampleSelect"
                  className='mt-3'
                  onChange={this.onNameChange} >
                  <option>seconds</option>
                  <option>minutes</option>
                  <option>hour(s)</option>
                  </Input>
                </Col>
                {errors.error || errors.cookingTime && <small style={{color: '#A43741' }}>{errors.cookingTime}</small>}
              </Col>
            </FormGroup>

            <FormGroup row>
              <Label for="exampleFile" sm={3}>File</Label>
              <Col sm={9}>
                <Input
                  type="file"
                  name="file"
                  id="exampleFile"
                  onChange={this.Upload}
                  accept="image/*"
                />
                <h6>{this.state.status}</h6>
                <FormText color="muted" />
              </Col>
            </FormGroup>

            <FormGroup check row >
              <Col sm={{ size: 10, offset: 2 }}>
                <Button onClick={this.onSubmit} style={{float: 'right', backgroundColor: '#A43741'}}>Add a recipe</Button>
              </Col>
            </FormGroup>
          </Form>
        </ModalBody>
      </Modal>
    );
  }
}


export default AddRecipeModal;
