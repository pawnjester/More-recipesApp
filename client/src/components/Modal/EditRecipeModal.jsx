import React, { Component } from 'react';
import superagent from 'superagent';
import PropTypes from 'prop-types';
import Dropzone from 'react-dropzone';
import { Button, Modal,
  ModalHeader, ModalBody, Form, Label, Input, FormGroup, Col, FormText } from 'reactstrap';
import sha1 from 'sha1';
import { connect } from 'react-redux';
import toastr from 'toastr';
import editRecipe from '../../actions/editRecipe';


class EditRecipeModal extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: '',
      ingredients: '',
      method: '',
      imageUrl: '',
      status: '',      
      id: this.props.id,
    };

    this.onNameChange = this.onNameChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.imageUpload = this.imageUpload.bind(this);
  }

  componentWillMount() {
    this.setState({
      name: this.props.recipe.name,
      ingredients: this.props.recipe.ingredients,
      method: this.props.recipe.method,
      imageUrl: this.props.recipe.imageUrl,
    });
  }

  onNameChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }


  onSubmit(event) {
    event.preventDefault();
    this.props.editRecipe(this.state);
    toastr.success('Recipe successfully changed');
    this.props.getAllRecipes();
    this.props.toggle();
  }

  imageUpload(files) {
    this.setState({ status: 'Uploading...' });
    const image = files[0];

    const cloudName = process.env.CLOUDNAME;

    const url = `https://api.cloudinary.com/v1_1/${cloudName}/image/upload`;

    const timestamp = Date.now() / 1000;
    const uploadPreset = process.env.UPLOADPRESET;

    const paramStr = `timestamp=${timestamp}&upload_preset=${uploadPreset}mr9XYhDjoiITDCHXlNm8jEvV03w`;

    const signature = sha1(paramStr);

    const params = {
      api_key: '569938115268323',
      timestamp,
      upload_preset: uploadPreset,
      signature,
    };
    const uploadRequest = superagent.post(url);
    uploadRequest.attach('file', image);
    Object.keys(params).forEach((key) => {
      uploadRequest.field(key, params[key]);
    });
    uploadRequest.end((err, resp) => {
      if (err) {
        toastr.success('Recipe cannot be editted');
        return;
      }

      this.setState({
        imageUrl: resp.body.secure_url,
        status: 'Uploaded',
      });
    });
  }
  render() {
    return (
      <Modal isOpen={this.props.isOpen} toggle={this.props.toggle}>
        <ModalHeader toggle={this.props.toggle}>Edit A recipe</ModalHeader>
        <ModalBody>
          <Form>
            <FormGroup row>
              <Label for="exampleEmail" sm={4}>Name</Label>
              <Col sm={8}>
                <Input
                  type="text"
                  name="name"
                  id="exampleEmail"
                  value={this.state.name}
                  onChange={this.onNameChange}
                  placeholder="Enter the name"
                />
              </Col>
            </FormGroup>

            <FormGroup row>
              <Label for="exampleEmail" sm={4}>Ingredients</Label>
              <Col sm={8}>
                <Input
                  type="text"
                  name="ingredients"
                  id="exampleEmail"
                  value={this.state.ingredients}
                  onChange={this.onNameChange}
                  placeholder="Enter the name"
                />
              </Col>
            </FormGroup>

            <FormGroup row>
              <Label for="exampleEmail" sm={4}>method</Label>
              <Col sm={8}>
                <Input
                  type="text"
                  name="method"
                  id="exampleEmail"
                  value={this.state.method}
                  onChange={this.onNameChange}
                  placeholder="Enter the name"
                />
              </Col>
            </FormGroup>

            <FormGroup row>
              <Label for="exampleFile" sm={4}>File</Label>
              <Col sm={8}>
                <Dropzone onDrop={this.imageUpload} />
                <h6>{ this.state.status}</h6>
                <FormText color="muted" />
              </Col>
            </FormGroup>

            <FormGroup check row>
              <Col sm={{ size: 10, offset: 2 }}>
                <Button onClick={this.onSubmit}>Edit a recipe</Button>
              </Col>
            </FormGroup>
          </Form>
        </ModalBody>
      </Modal>
    );
  }
}

EditRecipeModal.propTypes = {
  getAllRecipes: PropTypes.func.isRequired,

};


export default connect(null, { editRecipe })(EditRecipeModal);
