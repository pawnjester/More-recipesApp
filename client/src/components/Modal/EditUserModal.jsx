import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button, Modal,
  ModalHeader, ModalBody, Form, Label, Input, FormGroup, Col, FormText } from 'reactstrap';
import { connect } from 'react-redux';
import toastr from 'toastr';
import userDetail from '../../actions/editUserDetail';
import imageUpload from '../../helpers/imageUpload';


/* eslint-disable */

class EditUserModal extends Component {
  constructor(props) {
    super(props);

    this.state = {
      username: '',
      email: '',
      profileImg: '',
      status: '',
    };

    this.onNameChange = this.onNameChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.Upload = this.Upload.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.editUser) {
      this.setState({
        username: nextProps.editUser.username,
        email: nextProps.editUser.email,
        profileImg: nextProps.editUser.profileImg,
      });
    }
  }

  onNameChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  onSubmit(event) {
    event.preventDefault();
    this.props.userDetail(this.state);
    toastr.success('User successfully changed');
    this.props.getuserDetail();
    this.props.toggle();
  }

  Upload(images) {
    this.setState({ status: 'Uploading...'})
    imageUpload(images).then((response) => {
      const { body } = response
      const fileUrl = body.secure_url;

      if(fileUrl) {
        this.setState({
          profileImg: fileUrl,
          status: 'Uploaded'
        })
      }
    })
  }

  render() {
    const editedUser = (this.props.editUser) ? this.props.editUser : {};
    return (
      <Modal isOpen={this.props.isOpen} toggle={this.props.toggle}>
        <ModalHeader toggle={this.props.toggle} className="set-align">Edit A User</ModalHeader>
        <ModalBody>
          <Form>
            <FormGroup row>
              <Label for="exampleEmail" sm={4}>username</Label>
              <Col sm={8}>
                <Input
                  type="text"
                  name="username"
                  id="exampleEmail"
                  value={this.state.username}
                  onChange={this.onNameChange}
                  placeholder="Enter the name"
                />
              </Col>
            </FormGroup>

            <FormGroup row>
              <Label for="exampleEmail" sm={4}>email</Label>
              <Col sm={8}>
                <Input
                  type="text"
                  name="email"
                  id="exampleEmail"
                  value={this.state.email}
                  onChange={this.onNameChange}
                  placeholder="Enter the email"
                />
              </Col>
            </FormGroup>

            {/* <FormGroup row>
              <Label for="exampleEmail" sm={4}>method</Label>
              <Col sm={8}>
                <Input
                  type="text"
                  name="method"
                  id="exampleEmail"
                  value={this.state.method}
                  onChange={this.onNameChange}
                  placeholder="Enter the name"
                />
              </Col>
            </FormGroup> */}

            <FormGroup row>
              <Label for="exampleFile" sm={4}>File</Label>
              <Col sm={8}>
              <Input
                type="file"
                name="file"
                id="exampleFile"
                onChange={this.Upload}
                accept="image/*"
                />
                <h6>{ this.state.status}</h6>
                <FormText color="muted" />
              </Col>
            </FormGroup>

            <FormGroup check row>
              <Col sm={{ size: 10, offset: 2 }}>
                <Button onClick={this.onSubmit}>Edit profile</Button>
              </Col>
            </FormGroup>
          </Form>
        </ModalBody>
      </Modal>
    );
  }
}

EditUserModal.propTypes = {
  userDetail: PropTypes.func.isRequired,

};


export default connect(null, { userDetail })(EditUserModal);
