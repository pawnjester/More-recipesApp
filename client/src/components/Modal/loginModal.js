import React, { Component } from 'react';
import { Button, Modal, ModalHeader, ModalBody, ModalFooter, Form, FormGroup, Label, Input, FormText } from 'reactstrap';

class LoginModal extends Component {
  constructor(props) {
    super(props);
    this.state = {
      modal: false
    };

    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    this.setState({
      modal: !this.state.modal
    });
  }

  render() {
    return (
      <div>

      <Modal isOpen={this.state.modal} toggle={this.toggle} className={this.props.className}>
          <ModalHeader toggle={this.toggle}>Signup</ModalHeader>
          <ModalBody>
          <Form>
        <FormGroup>
          <Label for="exampleEmail">Email</Label>
          <Input type="email" name="email" id="exampleEmail" placeholder="Please enter your email address" />
        </FormGroup>

        <FormGroup>
          <Label for="examplePassword">Password</Label>
          <Input type="password" name="password" id="examplePassword" placeholder="Please enter your password" />
        </FormGroup>
        </Form>
            
          </ModalBody>
          <ModalFooter>
            <Button classNames = 'btn btn-outline-danger btn-block text-dark' onClick={this.toggle}>Do Something</Button>{' '}
          </ModalFooter>
        </Modal>

        
      </div>
    )
  }
}

Input.propTypes = {
  children: PropTypes.node,
  // type can be things like text, password, (typical input types) as well as select and textarea, providing children as you normally would to those.
  type: PropTypes.string,
  size: PropTypes.string,
  bsSize: PropTypes.valid,
  state: PropTypes.valid,
  valid: PropTypes.bool,
  tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
  // ref will only get you a reference to the Input component, use innerRef to get a reference to the DOM input (for things like focus management).
  innerRef: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
  static: PropTypes.plaintext,
  plaintext: PropTypes.bool,
  addon: PropTypes.bool,
  className: PropTypes.string,
  cssModule: PropTypes.object,
};

export default LoginModal
