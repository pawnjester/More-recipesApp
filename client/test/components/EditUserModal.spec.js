import React from 'React';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';
import sinon from 'sinon';
import { provider } from 'react-redux';
import { EditUserModal } from '../../src/components/Modal/EditUserModal';
import store from '../../src/store/store';

configure({ adapter: new Adapter() });

const props = {
  editUserDetail: jest.fn(),
  getuserDetail: jest.fn(),
  toggle: jest.fn(),
  errors: ''
}

function setup() {
  const shallowWrapper = shallow(<EditUserModal {...props} /> );
  return {
    shallowWrapper,
  }
}

describe('Test for Edit User Modal Modal', () => {
  it('should render correctly', () => {
    const { shallowWrapper } = setup();
    expect(shallowWrapper).toMatchSnapshot();
  });
});

describe('onChange() should', () => {
  it('be called when form input changes', () => {
    const { shallowWrapper } = setup();
    const event = {
      target: { name: 'username', value: 'phil' }
    };
    const inputForm = shallowWrapper.find('.name-field');
    inputForm.simulate('change', event);
    expect(shallowWrapper.instance().state.username).toBe('phil');
  });
});

describe('onSubmit() should', () => {
//   it('be called when form is submitted', () => {
//     const wrapper = shallow(<EditUserModal {...props} /> )
//     const event = {
//       preventDefault: jest.fn()
//     };
//     const form = wrapper.find('.edit-user-button');
//     wrapper.setState({ username: 'randomuser', profileImg: 'random.jpg' });
//     form.simulate('click', event);
//     expect(wrapper.instance().state.name).toBe('randomrecipe');
// });
  it('be called when form is submitted', () => {
    const wrapper = shallow(<EditUserModal {...props} />);
    const event = {
      preventDefault: jest.fn()
    };
    wrapper.setState({ username: 'randomuser', profileImg: 'random.jpg', errors: {} });
    const onSubmitSpy = jest.spyOn(
      wrapper.instance(), 'onSubmit'
    ).mockImplementation(() => Promise.reject({}))
    wrapper.instance().onSubmit(event);
    expect(onSubmitSpy).toHaveBeenCalled();
  });
});
