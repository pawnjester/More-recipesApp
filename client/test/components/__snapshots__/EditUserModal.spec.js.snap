// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Edit User Modal Modal should render correctly 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  fade={true}
  isOpen={false}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  role="dialog"
  toggle={[MockFunction]}
  zIndex={1050}
>
  <ModalHeader
    className="set-align"
    closeAriaLabel="Close"
    tag="h4"
    toggle={[MockFunction]}
    wrapTag="div"
  >
    Edit A User
  </ModalHeader>
  <ModalBody
    tag="div"
  >
    <Form
      tag="form"
    >
      <FormGroup
        row={true}
        tag="div"
      >
        <Label
          for="exampleEmail"
          sm={4}
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          username
        </Label>
        <Col
          sm={8}
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            className="name-field"
            id="exampleEmail"
            name="username"
            onChange={[Function]}
            placeholder="Enter the username"
            tag="p"
            type="text"
            value=""
          />
        </Col>
      </FormGroup>
      <FormGroup
        row={true}
        tag="div"
      >
        <Label
          for="exampleFile"
          sm={4}
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          File
        </Label>
        <Col
          sm={8}
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Input
            accept="image/*"
            id="exampleFile"
            name="file"
            onChange={[Function]}
            tag="p"
            type="file"
          />
          <h6 />
          <FormText
            color="muted"
            tag="small"
          />
        </Col>
      </FormGroup>
      <FormGroup
        check={true}
        row={true}
        tag="div"
      >
        <Col
          sm={
            Object {
              "offset": 2,
              "size": 10,
            }
          }
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Button
            className="edit-user-button"
            color="secondary"
            onClick={[Function]}
            tag="button"
          >
            Edit profile
          </Button>
        </Col>
      </FormGroup>
    </Form>
  </ModalBody>
</Modal>
`;
